internal/cmd/gentypes/main.go

```diff
@@ -146,6 +147,9 @@ func run(_ *cobra.Command, args []string) {
 				fmt.Fprintf(w, "\nstate.ChainHeader\n")
 			}
 		}
+		for _, e := range typ.Embeddings {
+			fmt.Fprintf(w, "\t\t%s\n", e)
+		}
```

Comment:
```suggestion
		if len(typ.Embeddings) > 0 && !typ.NonBinary {
			check(fmt.Errorf("type %q: embedding is not supported for binary-marshalled types", typ.name))
		}
		for _, e := range typ.Embeddings {
			fmt.Fprintf(w, "\t\t%s\n", e)
		}
```

The binary marshal generator does not know about embedded fields, so until it does, we need to tell the user that is not supported.

Reply:
Done (in next commit)

Looking at this code I am also inclined to refactor like this:
```
printEmbeddings(typ, w) 
printFields(typ, w) 
fmt.Fprintf(w, "}\n\n") 
```
But that will give a much larger PR and gentypes is not really in scope of this ticket to begin with.

Reply:
The type generator could definitely use some love